src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -100,8 +115,78 @@ class AmendmentTableImpl : public AmendmentTable
 void
 AmendmentTableImpl::addInitial ()
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+    std::set<std::pair<uint256, std::string>> toAdd;
+
+    for (auto const& _ : detail::PreEnabledAmendments)
+    {
+        uint256 hash;
+        if (!hash.SetHex (_.first))
+        {
+            std::string const errorMsg =
+                (boost::format (
+                     "PreEnabledAmendments contains an invalid hash (expected "
+                     "a hex number). Value was: %1%") %
+                 _.first).str ();
+            throw std::runtime_error (errorMsg);
```

Comment:
We're trying to get away from using `boost::format` though this isn't a terrible use compared to some.  Indeed, you might not need the intermediate variable:

    throw std::runtime_error ( "PreEnabledAmendments contains an invalid hash (expected "
                     "a hex number). Value was: " + to_string(a.first));